content/v1.17/guides/extensions-release-process.md

```diff
@@ -0,0 +1,352 @@
+instance with APIs and business logic for composing resources.
+
+Building a Crossplane extension involves creating OCI images in the [xpkg]
+format. Authors and maintainers of Crossplane extensions must push their
+packages to an OCI registry before users can reference and use them.
+
+The release process for Crossplane extensions grew organically in the community
+and developed its own conventions and common configurations. Authors of these
+extensions should follow this guide to enable automation for building
+and pushing their packages as part of their git workflow.
+
+This guide provides step-by-step instructions for configuring automated
+CI pipelines in GitHub Actions for pushing your Crossplane extensions to
+`xpkg.crossplane.io` and/or `xpkg.upbound.io`, which are two registries
+that the Crossplane community use today.
+
+{{< hint "tip" >}}
+For more information about Crossplane packages, review the
+[xpkg concepts]({{<ref "../../master/concepts/packages" >}}).
+{{< /hint >}}
+
+## Typical workflow
+
+A typical GitHub workflow definition contains the following steps:
+
+1. Fetching the source repository
+2. Authenticating to a remote registry
+3. Building and packaging artifacts
+4. Pushing (publishing) the artifact
+
+{{< hint "warning" >}}
+The supplied credentials for the remote registry require read and write access
+as upload requests to the registry specify `push` authorization scope.
+{{< /hint >}}
+
+## Quickstart: Releasing a Provider to `xpkg.crossplane.io`
+
+### Prerequisites
+
+- A GitHub repository, for example created from the
+[Upjet template](https://github.com/crossplane/upjet-provider-template)
+
+### Steps
+
+1. Create a new YAML file under `.github/workflows`. By convention, name this
+file `publish-provider-package.yaml`.
+2. Copy the following workflow definition into the file, replacing
+`<REPOSITORY NAME>` with the desired name of the repository in the registry.
+
+    ```yaml
+    name: Publish Provider Package
+
+    on:
+      workflow_dispatch:
+        inputs:
+          version:
+            description: "Version string to use while publishing the package (e.g. v1.0.0-alpha.1)"
+            default: ''
+            required: false
+          go-version:
+            description: 'Go version to use if building needs to be done'
+            default: '1.23'
+            required: false
+
+    jobs:
+      publish-provider-package:
+        uses: crossplane-contrib/provider-workflows/.github/workflows/publish-provider-non-family.yml@main
+        with:
+          repository: <REPOSITORY NAME>
+          version: ${{ github.event.inputs.version }}
+          go-version: ${{ github.event.inputs.go-version }}
+          cleanup-disk: true
+        secrets:
+          GHCR_PAT: ${{ secrets.GITHUB_TOKEN }}
+    ```
+
+3. Commit the workflow file to the default branch of the GitHub repository.
+4. The workflow should now be available to trigger via the GitHub UI in the
```

Comment:
something that still feels missing is the choices/actions around the release itself. do you feel like those are in scope here too? e.g. creating a release branch and/or tagging the release version. 

or perhaps it could be as simple as adding here `...when you are ready to publish a release of your provider` 😇

Reply:
beyond that though, I'm still personally struggling with the inconsistency we have introduced over time into the provider ecosystem. e.g. the ci.yaml for upjet-provider-template publishes on every CI run: https://github.com/crossplane/upjet-provider-template/blob/main/.github/workflows/ci.yml#L320-L322. It's sort of odd that we now have a separate workflow that is manually triggered, instead of just letting ci.yml publishing do the publishing itself.

This first came up in https://github.com/crossplane/org/issues/88, we're sort of fractured the ecosystem into different paths that folks have to figure out themselves. is there much we can do in these docs to help people understand a single easy path to take? 🤔

Reply:
> fractured the ecosystem into different paths

completely agree here -- this became obvious as I was writing these docs. my understanding is that the reusable workflows are meant to be the desired "golden path" that requires fewer changes to Makefiles and other subtle configuration (cc @jeanduplessis @sergenyalcin to keep me honest here). I'm highlighting that path in these docs, but a follow-up would be to update the provider templates to include the reusable workflow and deprecate the publishing portion from `ci.yml`.